lib/src/main/java/me/onebone/toolbar/CollapsingToolbar.kt

```diff
@@ -136,8 +125,23 @@ class CollapsingToolbarState(
 	)
 	fun feedScroll(value: Float): Float = dispatchRawDelta(value)
 
+	/**
+	 * @return Remaining velocity after fling
+	 */
+	suspend fun fling(flingBehavior: FlingBehavior, velocity: Float): Float {
+		var left = velocity
+		scroll {
+			with(flingBehavior) {
+				left = performFling(left)
+			}
+		}
+
+		return left
+	}
+
+	// TODO: A strange jump in snap speed is often observed
```

Comment:
Can you explain this issue?

Reply:
This is a bug I mentioned. You can see a strange expanding before the snap will applied. This sometimes works even in wrong direction. Look at the demonstration GIF:
<img src="https://user-images.githubusercontent.com/7997602/150323590-52614361-5e96-4436-a595-c552ebcb6b40.gif" width="300px">

When I release the finger you can see that snapping is moving in wrong direction (to top) before it finally snap to full size.

I tried to debug it and I suppose that a bug is in implementation of `EnterAlwaysCollapsedNestedScrollConnection#onPostScroll()` method. Unfortunately I don't know how `NestedScrollConnection` and math in `EnterAlwaysCollapsedNestedScrollConnection` work. I will be glad if you help me with it. Try to build an example app and reproduce this bug.